src/bin/snx-rs.rs

```diff
@@ -140,3 +146,35 @@ async fn main() -> anyhow::Result<()> {
 
     result
 }
+
+async fn get_server_info(params: &TunnelParams) -> anyhow::Result<ServerInfoResponse> {
+    let client = CccHttpClient::new(Arc::new(params.clone()), None);
+    let info = client.get_server_info().await?;
+    let response_data = info.get("ResponseData").unwrap_or(&Value::Null);
+    Ok(serde_json::from_value::<ServerInfoResponse>(response_data.clone())?)
+}
+
+async fn get_server_pwd_prompts(params: &TunnelParams) -> anyhow::Result<VecDeque<String>> {
+    let mut pwd_prompts = VecDeque::new();
+    if !params.server_prompt {
+        return Ok(pwd_prompts);
+    }
+    let server_info = get_server_info(params).await?;
+    let login_type = &params.login_type;
+    let login_factors = server_info
+        .login_options_data
+        .login_options_list
+        .iter()
+        .find(|login_option| login_option.id == *login_type)
+        .map(|login_option| login_option.to_owned())
+        .unwrap()
```

Comment:
Please avoid unwraps, in this particular case it could be:
```rust
let login_factors = server_info
        .login_options_data
        .login_options_list
        .into_iter()
        .find_map(|login_option| if login_option.id == *login_type { login_option.factors } else { None })
        .unwrap_or_default()
```

Reply:
Sure, don't know what made me use it here, refactored that part.